File: pdlearning/src/microservices/calendar-microservice/Api/Application/Models/UserAccessSharingModel.cs
using System;

namespace Microservice.Calendar.Application.Models
{
    public class UserAccessSharingModel
    {
        public Guid UserId { get; set; }

        public string FullName { get; set; }

        public string Email { get; set; }

        public bool Shared { get; set; }
    }
}

File: pdlearning/src/web-apps/src/libraries/domain-api/src/calendar/models/user-access-sharing-model.ts
export interface IUserAccessSharingModel {
  userId: string;
  fullName: string;
  email: string;
  shared: boolean;
}

export class UserAccessSharingModel implements IUserAccessSharingModel {
  public userId: string;
  public fullName: string;
  public email: string;
  public shared: boolean;

  constructor(data: IUserAccessSharingModel) {
    this.userId = data.userId;
    this.fullName = data.fullName;
    this.email = data.email;
    this.shared = data.shared;
  }
}
